edb/server/protocol/auth_ext/http.py

```diff
@@ -589,13 +596,16 @@ async def handle_request(
                                 query, 'error'
                             ),
                             email=_maybe_get_search_param(query, 'email'),
+                            challenge=_get_search_param(
+                                query, 'challenge'
+                            ),
```

Comment:
This breaks the "Sign in" link on the "Sign up" page.

Reply:
ah, yeah, I can pass it to sign in as well so it can pass them on. I originally did that assuming we had the social buttons there too. Probably want to add them?

Reply:
Maybe we could put it in a session cookie? So if that 'challenge' query param exists, we use it and put it in a session cookie, if not then we check the cookie. Then we can also handle the less secure flow, by checking if neither exist, then we create a new verifier and put that in a session cookie, and use that to derive the challenge when we need it.

Reply:
That's a great idea @jaclarke 🙌

Reply:
@jaclarke @fantix ⚡ [`6e84957` (#6239)](https://github.com/edgedb/edgedb/pull/6239/commits/6e84957532ac2be1a796ac004deae708477abd8b)

Gotta love writing cookie parsers from scratch 😅